test/src/frame.spec.ts

```diff
@@ -138,72 +168,102 @@ describe('Frame specs', function () {
     it('should not send attach/detach events for main frame', async () => {
       const {page, server} = await getTestState();
 
-      let hasEvents = false;
-      page.on('frameattached', () => {
-        return (hasEvents = true);
+      const frameAttached = waitForCondition(
+        page,
+        'frameattached',
+        () => {
+          return true;
+        },
+        200
+      ).catch(() => {
+        return false;
       });
-      page.on('framedetached', () => {
-        return (hasEvents = true);
+      const frameDetached = waitForCondition(
```

Comment:
I am not sure if we need to change the tests here. We test Puppeteer's logic and the expectation is that Puppeteer will emit detached events on navigation: see #removeFramesRecursively in #onFrameNavigated.

Reply:
As discussed offline, we will investigate whether we try to detect this condition another way.

References:

 - https://github.com/w3c/webdriver-bidi/issues/659